src/main/java/bio/terra/app/controller/SnapshotsApiController.java

```diff
@@ -146,7 +146,7 @@ public ResponseEntity<JobModel> createSnapshot(
       // we can retrieve the job we just created
       return jobToResponse(jobService.retrieveJob(jobId, userReq));
     }
-    throw new IamUnauthorizedException(
+    throw new IamForbiddenException(
         "User is not authorized to create snapshots for these datasets " + unauthorized);
```

Comment:
Not part of the changes made in this commit, but is this Controller leaking the fact that a dataset exists when the user is forbidden?

Reply:
Yeah, seemed like it. I think there was a comment on one of the tests to fix that.